src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts

```diff
@@ -1,61 +1,94 @@
 import { html } from "lit-html";
 import { asyncReplace } from "lit-html/directives/async-replace.js";
+import { createRef, ref, Ref } from "lit-html/directives/ref.js";
+import type QrCreator from "qr-creator"; // XXX: import to only import the _type_ to avoid pulling in the whole module (module itself is used as a dynamic import)
 import {
   DeviceData,
   Timestamp,
 } from "../../../../generated/internet_identity_types";
+import { checkmarkIcon, copyIcon } from "../../../components/icons";
 import { mainWindow } from "../../../components/mainWindow";
+import { toast } from "../../../components/toast";
 import { addDeviceLink } from "../../../utils/addDeviceLink";
 import { AsyncCountdown } from "../../../utils/countdown";
 import { AuthenticatedConnection } from "../../../utils/iiConnection";
-import { renderPage } from "../../../utils/lit-html";
+import { mount, renderPage, withRef } from "../../../utils/lit-html";
 import { delayMillis } from "../../../utils/utils";
 
 const pollForTentativeDeviceTemplate = ({
+  userNumber,
   cancel,
   remaining,
-  link,
+  origin,
 }: {
+  userNumber: bigint;
   cancel: () => void;
   remaining: AsyncIterable<string>;
-  link: string;
+  origin: string;
 }) => {
+  const link = addDeviceLink({ userNumber, origin });
+  const copyLink_ = () => navigator.clipboard.writeText(link);
+
+  const linkCopyElement: Ref<HTMLElement> = createRef();
+  // Copy the link and give visual feedback on success
+  const copyLink = async () => {
+    try {
+      await copyLink_();
+      withRef(linkCopyElement, (linkCopyElement) => {
+        linkCopyElement.classList.add("is-copied");
+      });
+    } catch (e: unknown) {
+      toast.error("Could not copy link to clipboard");
```

Comment:
I think that would make sense but (see here) sometimes it's interesting to have something a bit more user friendly in the toast. wdyt?

Reply:
Yeah definitely. In NNS-dapp we print a user friendly message in the toast + interpretation of the err + console.error, everything through the same function.

Anyhow not related to this PR but I think would be interesting to avoid repeating `console.error`